app/javascript/src/hyacinth_v1/components/projects/enabled_dynamic_fields/enabled_dynamic_field_form/EnabledDynamicFieldForm.js

```diff
@@ -0,0 +1,283 @@
+import { useMutation, useQuery } from '@apollo/react-hooks';
+import produce from 'immer';
+import pick from 'lodash/pick';
+import PropTypes from 'prop-types';
+import React, { useReducer, useState } from 'react';
+import { Card, Form } from 'react-bootstrap';
+import { useHistory } from 'react-router-dom';
+import { getDynamicFieldGraphQuery } from '../../../../graphql/dynamicFieldCategories';
+import { getProjectFieldSetsQuery } from '../../../../graphql/projects';
+import { getEnabledDynamicFieldsQuery, updateEnabledDynamicFieldsMutation } from '../../../../graphql/projects/enabledDynamicFields';
+import ErrorList from '../../../shared/ErrorList';
+import FormButtons from '../../../shared/forms/FormButtons';
+import GraphQLErrors from '../../../shared/GraphQLErrors';
+import EnabledDynamicField from './EnabledDynamicField';
+
+const findDynamicFields = (searchTarget, dynamicFields = []) => {
+  if (searchTarget instanceof Array) {
+    searchTarget.forEach((el) => findDynamicFields(el, dynamicFields));
+  } else if (searchTarget instanceof Object) {
+    if (searchTarget.type === 'DynamicField') {
+      dynamicFields.push(searchTarget);
+    } else {
+      Object.values(searchTarget).forEach((val) => {
+        findDynamicFields(val, dynamicFields);
+      });
+    }
+  }
+  return dynamicFields;
+};
+
+const edfDefaultState = { enabledFieldHierarchy: [] };
+const edfReducer = (state, action) => {
```

Comment:
if this is truly a function (that is, it does not refer to names in current scope), might want to consider a clean-up ticket to extract into a module.

Reply:
Great suggestion!  Done for `edfReducer` and also `findDynamicFields`!